### PR TITLE
Fun test improvements

### PR DESCRIPTION
**Pull request by mnonnenmacher**
See the commit messages for details.

### CODE REVIEW

File: analyzer/src/funTest/kotlin/managers/DotNetTest.kt [hunk 1]
```diff
@@ -44,7 +46,10 @@ class DotNetTest : StringSpec() {
             val mapper = DotNetPackageReferenceMapper()
             val result = mapper.mapPackageReferences(packageFile)
 
-            result.size shouldBe 2
+            result should containExactly(mapOf(
```
Comment (sschuberth) [inline analyzer/src/funTest/kotlin/managers/DotNetTest.kt:49]: In the one other location with similar code

https://github.com/oss-review-toolkit/ort/blob/bd88d4bba90c3ed444555fe651954f15d58a3a8b/utils/src/test/kotlin/DeclaredLicenseProcessorTest.kt#L64

we use the `shouldContainExactly` single-word variant. Have we decided yet on which to align? Would you mind making an alignment commit and also remove the superfluous space before the parenthesis in the above code?

Comment (mnonnenmacher) [reply analyzer/src/funTest/kotlin/managers/DotNetTest.kt:49]: For collections there is a vararg variant of `should containExactly` but not of `shouldContainExactly`, that's why I would prefer to align on the former. For maps there is no vararg variant, but we could make our own if we want. `should containExactly` is also dominant in the code base and better aligned with most other matchers we use. I'll make the alignment commit.

File: test-utils/build.gradle.kts [hunk 2]
```diff
@@ -30,6 +30,7 @@ dependencies {
     api(project(":model"))
 
     api("io.kotest:kotest-core-jvm:$kotestVersion")
+    api("io.kotest:kotest-assertions-core-jvm:$kotestVersion")
```
Comment (sschuberth) [inline test-utils/build.gradle.kts:33]: Does it make sense to also file this as a feature request for upstream kotest?

File: analyzer/src/test/kotlin/managers/utils/NodeSupportTest.kt [hunk 3]
```diff
@@ -272,6 +283,12 @@ class NodeSupportTest : WordSpec() {
 
     private fun absolutePaths(vararg files: String) = files.map { tempDir.resolve(it) }
 
+    private fun containAbsolutePathsExactly(vararg paths: String): Matcher<Collection<File>?> =
+        containExactly(absolutePaths(*paths) as Collection<File>)
+
+    private fun containAbsolutePathsExactlyInAnyOrder(vararg paths: String): Matcher<Collection<File>?> =
+        containExactlyInAnyOrder(absolutePaths(*paths))
+
```
Comment (sschuberth) [inline analyzer/src/test/kotlin/managers/utils/NodeSupportTest.kt:292]: Hmm. I have do admit I do not like this change. It doesn't really increase readability significantly, at the cost of introducing very specialized functions. How maybe other permutations of `shouldContainExactly(InAnyOrder) <helper function>` do we want to wrap in yet another helper function.

I'd prefer to drop this commit.

File: test-utils/build.gradle.kts (hunk 2 above)
Comment (mnonnenmacher) [reply test-utils/build.gradle.kts:33]: Yes, I'm already looking into it.

File: analyzer/src/test/kotlin/managers/utils/NodeSupportTest.kt (hunk 3 above)
Comment (mnonnenmacher) [reply analyzer/src/test/kotlin/managers/utils/NodeSupportTest.kt]: Removed it.

Comment (sschuberth) [reply analyzer/src/test/kotlin/managers/utils/NodeSupportTest.kt]: Thanks.

File: analyzer/src/test/kotlin/managers/utils/NodeSupportTest.kt [hunk 4]
```diff
@@ -270,7 +276,7 @@ class NodeSupportTest : WordSpec() {
         if (hasYarnLockFile) projectDir.resolve("yarn.lock").createNewFile()
     }
 
-    private fun absolutePaths(vararg files: String) = files.map { tempDir.resolve(it) }
+    private fun absolutePaths(vararg files: String): Collection<File> = files.map { tempDir.resolve(it) }
```
Comment (sschuberth) [inline analyzer/src/test/kotlin/managers/utils/NodeSupportTest.kt:279]: Is this actually required or just "nice to have"? Should it maybe to into the `NodeSupportTest: Simplify 'absolutePaths'` commit instead?

File: analyzer/src/test/kotlin/managers/utils/NodeSupportTest.kt [hunk 5]
```diff
@@ -198,14 +204,14 @@ class NodeSupportTest : WordSpec() {
                 fun ProtocolProxyMap.mapSingleValuesToString() =
                     mapValues { (_, proxies) ->
                         val (proxy, authentication) = proxies.single()
-                        listOf(proxy.toString(), authentication?.userName, authentication?.password?.let { String(it) })
+                        listOf(proxy.toString(), authentication!!.userName, String(authentication.password))
```
Comment (sschuberth) [inline analyzer/src/test/kotlin/managers/utils/NodeSupportTest.kt:207]: This change seems unrelated, and the simplification also is not correct, as `authentication` *can* be `null`. I was just missing a test case for that, but I've added it now: https://github.com/oss-review-toolkit/ort/pull/2750. That new est would result in an error with your change.

Comment (mnonnenmacher) [reply analyzer/src/test/kotlin/managers/utils/NodeSupportTest.kt]: The simplification was correct before your change, and was required because the type resolver had issues determining which matcher to use. But I have dropped all changes in this file from the last commit for that reason.

File: analyzer/src/test/kotlin/managers/utils/NodeSupportTest.kt (hunk 4 above)
Comment (mnonnenmacher) [reply analyzer/src/test/kotlin/managers/utils/NodeSupportTest.kt]: See below.

File: analyzer/src/test/kotlin/managers/utils/NodeSupportTest.kt (hunk 5 above)
Comment (sschuberth) [reply analyzer/src/test/kotlin/managers/utils/NodeSupportTest.kt]: Does this [commit](https://github.com/oss-review-toolkit/ort/pull/2750/commits/8bea7c2e855657a380d8bb0955fecad3d94527a5) I just added to my PR help in that regard?

Comment (mnonnenmacher) [reply analyzer/src/test/kotlin/managers/utils/NodeSupportTest.kt]: IIRC there were more issues than just the nullability, but I'm not sure. I'd suggest to give it a try as part of your commit.

Comment (sschuberth) [reply analyzer/src/test/kotlin/managers/utils/NodeSupportTest.kt]: I've incorporated my changes to this PR and restored your changes to `NodeSupportTest.kt` in the last commit.
